### ISSUE TITLE
stream to cloud function using webusb

### ISSUE THREAD

**Issue by machinehistories** [microsoft/pxt-microbit#1357]
I am requesting the previously capable stream to cloud feature is reinstated to allow data from microbit to be streamed to google sheets or microsoft azure. It would be great to see a set of blocks that enable a pc iot cloud gateway and can make an https get request so we can use services such as adafruit.io using . There are several add ons for the microbit that feature esp8266 chip and they do this sort of thing already by using wifi to serial 
[https://github.com/alfonsodambrosio/microbitwifi2](url)
[https://www.dfrobot.com/wiki/index.php/Gravity:_UART_OBLOQ_IoT_Module_SKU:_TEL0118](url)
[https://github.com/elecfreaks/pxt-esp8266iot](url)
I am requesting either webusb or webbluetooth is used to achieve a similar result which is to upload data to a rest or mqtt APIs such as thingspeak, adafruitio, azure, etc. I believe several of the blocks listed in the links above could be re purposed towards these ends with the help of a websocket or somesuch method. 
The goal of this request is for users to be able to send data from the microbit and store data in the cloud and to also be able to send data from the cloud to the microbit in order to control pins and events.

**Comment by pelikhan** [comment]
We should be able to implement this kind of features as a editor extension. I'll prep a sample.

**Comment by machinehistories** [comment]
Amazing I can't wait to use it. I never got an opportunity to use the
previous stream to cloud function. I did cobble together my own blocks for
esp8266 with microbit for use with adafruit.io. I took some of the things
that worked with obloq and some with elecfreaks extension and
Frankensteined them. I know you are going to do something far more polished
without the need for external hardware. With all the web protocols you are
incorporating into makecode the microbit can be a hub for all sorts of
events. Have you played with
https://hourofpython.trinket.io/from-blocks-to-code-with-trinket#/blocks/dragging-and-dropping.
I imagine adding some vector graphics rendering/programming controlled by
the microbit (similar to scratch) could be another very cool avenue and
might be a neat extension as well. Sorry I am not being greedy or
suggesting you do this just sharing a link. I'll just leave it with a thank
you for your hard work. - jason
